index.html

```diff
@@ -1802,6 +1950,122 @@ <h4 class="modal-title">Help</h4>
                     console.log('right up');
                     this.remove_total_highlight();
                 },
+                //bring pieces to original positions
+                resetpieces: function() {
+                    this.highlighted = null;
+                    this.selected = null;
+                    this.selectedStack = null;
+                    this.move = {start: null, end: null, dir: 'U', squares: []};
+
+                    //reset stacks
+                    var piece;
+                    var whitePieces = [];
+                    var whiteCaps = [];
+                    var blackPieces = [];
+                    var blackCaps = [];
+                    while (piece = this.piece_objects.pop())
+                    {
+                      if (piece.onsquare)
+                        continue;
+                      this.flatten(piece);
+                      if (piece.iswhitepiece) {
+                        if (piece.iscapstone)
+                          whiteCaps.push(piece);
+                        else
+                          whitePieces.push(piece);
+                      }
+                      else {
+                        if (piece.iscapstone)
+                          blackCaps.push(piece);
+                        else
+                          blackPieces.push(piece);
+                      }
+                    }
+                    for (var i = 0; i < this.size; i++) {
+                        for (var j = 0; j < this.size; j++) {
+                            while (piece = this.sq[i][j].pop()) {
```

Comment:
This is where board.sq gets 'reset'. The elements of each square are being removed from the array.
Thus, the array is empty when the pieces get arranged.